VerilogCodingStyle.md

```diff
@@ -1928,26 +2003,65 @@ always_comb begin
     3'b010: operand = accum1 >> 0;
     3'b011: operand = accum1 >> 1;
     3'b1??: operand = regfile[select[1:0]];
-    default: operand = 'X;  // propagate X
+    default: operand = '0; // assign a default
   endcase
 end
 ```
 
 The `unique` prefix is recommended before all case statements, as it creates
 simulation assertions that can catch certain mistakes. In some cases, `priority`
-may be used instead of `unique`, though `if-else-if` structures are a more
-readable representation for priority encoders.
-
-Be sure to use `unique case` correctly. In particular, any variables assigned in
-one case item must be assigned in all case items, including the default. Failing
-to do this can lead to a simulation-synthesis mismatch as described in [Don
-Mills' paper][yalagp].
+may be used instead of `unique`, though in such cases, cascaded ternary
+structures should be the preferred way of representing priority encoders as
+they are a more readable representation for priority encoders.
+
+Be sure to use `unique case` correctly. In particular, make sure that:
+
+  - a `default:` statement is **always** included in order to avoid accidental
+  inference of latches, even if all cases are covered. In simulation, a case
+  expression that evaluates to `X` will not match any case and will behave as a
+  latch, leading to different behavior than synthesis if no default is specified.
+
+  - if no default assignments are given before the case statement as shown in
+  the example above, any variables assigned in one case item must be assigned in
+  all case items, including the `default:`. Failing to do this can lead to a
+  simulation-synthesis mismatch as described in [Don Mills' paper][yalagp].
+
+The following is a different example showing a style-compliant case statement
```

Comment:
@sjgitty shall I omit this second case statement example? looks like this is not strictly needed after stating the points above...

Reply:
I think it is fine as written now, thanks.